org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/fix/MissingAnnotationAttributesFixCore.java

```diff
@@ -0,0 +1,49 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.internal.corext.fix;
+
+import org.eclipse.core.runtime.Assert;
+
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Annotation;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+
+import org.eclipse.jdt.internal.ui.text.correction.CorrectionMessages;
+import org.eclipse.jdt.internal.ui.text.correction.IProblemLocationCore;
+
+public class MissingAnnotationAttributesFixCore extends CompilationUnitRewriteOperationsFixCore {
+
+	public MissingAnnotationAttributesFixCore(String name, CompilationUnit compilationUnit, CompilationUnitRewriteOperation operation) {
+		super(name, compilationUnit, operation);
+	}
+
+	/**
+	 * Returns a fix that adds the missing required attributes to the annotation at the problem
+	 * location.
+	 *
+	 * @param cu the AST of the compilation unit where an attribute is missing required attributes
+	 * @param problemLocation the error marker that marks the annotation that's missing attributes
+	 * @return a fix that adds the missing required attributes to the annotation at the problem
+	 *         location
+	 */
+	public static MissingAnnotationAttributesFixCore addMissingAnnotationAttributesProposal(CompilationUnit cu, IProblemLocationCore problemLocation) {
+		ASTNode selectedNode= problemLocation.getCoveringNode(cu);
+		Assert.isTrue(selectedNode instanceof Annotation);
```

Comment:
Do we need this (when we eventually implement it in a language server, or some core component) ? Could we just return `null` and check for before creating the proposal.

Reply:
Yeah, that makes sense to me. I'll update it.